gateway/src/resty/resolver.lua

```diff
@@ -98,10 +101,17 @@ function _M.parse_nameservers(path)
   end
 
   if resolver then
-    local m = re.split(resolver, ':', 'oj')
-    insert(nameservers, nameserver.new(m[1], m[2]))
-    -- we are going to use all resolvers, because we can't trust dnsmasq
-    -- see https://github.com/3scale/apicast/issues/321 for more details
+    local m
+
+    m, err = re_match(resolver, '^(.+?)(?<!:)(?:\\:(\\d+))?$', 'oj')
```

Comment:
If we extract the regex to a constant, readers will be able what we need to match here without understanding the regex.

Reply:
It matches anything optionally followed by just one `:` and a series of numbers.
I'm not even sure how to call it :) 

It needs to match `1.2.3.4:53` and `[dead::beef]:53` but not `dead::beef::53`.

Actually I see there is a bug as it matches the 3rd example as a port. http://rubular.com/r/j6rm3ObFIM

Reply:
Updated to extended regex with comments https://github.com/3scale/apicast/pull/511/commits/9226b0bee3f65dc178f198b401b482992df9dd02